extensions/dust-tt/src/pickWorkspace.tsx

```diff
@@ -2,102 +2,165 @@ import {
+        if (!user) {
+          user = euResult.value;
+        }
       }
-      return undefined;
-    },
-    [isUsingOauth, isLoadingUser],
-  );
+    } catch (error) {
+      // The user might not be exist in EU region, so we can ignore errors here.
+    }
+
+    // Store user info for greeting
+    if (user) {
+      await LocalStorage.setItem("user", JSON.stringify(user));
+    }
+
+    return { usWorkspaces, euWorkspaces, user };
+  }, []);
+
+  const { data: currentWorkspaceId, isLoading: isLoadingWorkspace } = usePromise(async () => getWorkspaceId(), []);
 
   const setWorkspaceId = useCallback(
-    async (workspace: { sId: string; name: string }) => {
-      updateCommandMetadata({ subtitle: `Currently using: "${workspace.name}"` });
+    async (workspace: WorkspaceWithRegion) => {
+      updateCommandMetadata({ subtitle: `Currently using: "${workspace.name}" (${workspace.region})` });
       await LocalStorage.setItem("workspaceId", workspace.sId);
+      await LocalStorage.setItem("selectedRegion", workspace.region);
       navigation.pop();
     },
     [navigation],
   );
 
-  if (isUsingOauth) {
-    return (
-      <List isLoading={isLoadingUser || isLoadingWorkspace} selectedItemId={workspaceId}>
-        {user && (
-          <List.Section title={`${getRandomGreetingForName(user.firstName)}, please pick a workspace below:`}>
-            {user.workspaces.map((workspace) => (
-              <List.Item
-                key={workspace.sId}
-                id={workspace.sId}
-                title={workspace.name}
-                subtitle={workspace.sId}
-                accessories={[
-                  {
-                    tag: { value: workspace.role, color: workspace.role === "admin" ? Color.Blue : Color.Green },
-                    icon: workspaceId === workspace.sId ? { source: Icon.Check } : null,
-                  },
-                ]}
-                actions={
-                  <ActionPanel>
-                    <Action
-                      title="Select"
-                      icon={Icon.Check}
-                      shortcut={{ key: "return", modifiers: [] }}
-                      onAction={() => setWorkspaceId(workspace)}
-                    />
-                  </ActionPanel>
-                }
-              />
-            ))}
-          </List.Section>
-        )}
-      </List>
-    );
-  } else {
-    return (
-      <Detail
-        navigationTitle="Not using OAuth"
-        markdown="Workspace selection is **only available when connecting with your own Dust account** and not **workspace API Key**. _You can change the connexion method in the extension preferences._"
-      />
-    );
-  }
+  const user = workspacesData?.user;
+  const usWorkspaces = workspacesData?.usWorkspaces || [];
+  const euWorkspaces = workspacesData?.euWorkspaces || [];
+
+  return (
+    <List isLoading={isLoadingWorkspaces || isLoadingWorkspace} selectedItemId={currentWorkspaceId}>
```

Comment:
I think we should basically rely on the workspace that was selected in workos (selectedWorkspace) ?